shared/js/background/declarative-net-request.js

```diff
@@ -327,6 +331,16 @@ export async function refreshUserAllowlistRules (allowlistedDomains) {
     await updateUserAllowlistRule(normalizedAllowlistedDomains)
 }
 
+export async function flushSessionRules () {
```

Comment:
IMO let's move this into `dnr-session-rule-id.js`. Perhaps we could rename stuff in that module a bit to make it clear the distinction between session rules with incremented ID and other session rules too?

Reply:
Not sure I follow, `dnr-session-rule-id.js` seems to be focused on managing rule IDs, while `declarative-net-request.js` is about managing the rules themselves (eg. `toggleUserAllowlistDomain`) which is what `flushSessionRules` does?